packages/forgetti/runtime/index.ts

```diff
@@ -1,19 +1,51 @@
+function isEqual(a: unknown, b: unknown): boolean {
+  // eslint-disable-next-line no-self-compare
+  return a === b || (a !== a && b !== b);
+}
+
 export function $$equals(cache: unknown[], index: number, b: unknown): boolean {
-  if (index in cache) {
-    const a = cache[index];
-    // eslint-disable-next-line no-self-compare
-    return a === b || (a !== a && b !== b);
-  }
-  return false;
+  return index in cache && isEqual(cache[index], b);
```

Comment:
`$$equals` is a very hot path, and using a function call here (which adds a stack) could cause overhead. [Benchmark](https://perf.link/#eyJpZCI6Imt1YnowYTlvbm4xIiwidGl0bGUiOiIiLCJiZWZvcmUiOiJmdW5jdGlvbiAkJGVxdWFsczEoY2FjaGUsIGluZGV4LCBiKSB7XG4gIGlmIChpbmRleCBpbiBjYWNoZSkge1xuICAgIGNvbnN0IGEgPSBjYWNoZVtpbmRleF07XG4gICAgcmV0dXJuIGEgPT09IGIgfHwgKGEgIT09IGEgJiYgYiAhPT0gYik7IFxuICB9XG4gIHJldHVybiBmYWxzZTtcbn1cblxuZnVuY3Rpb24gaXNFcXVhbChhLCBiKSB7XG4gIC8vIGVzbGludC1kaXNhYmxlLW5leHQtbGluZSBuby1zZWxmLWNvbXBhcmVcbiAgcmV0dXJuIGEgPT09IGIgfHwgKGEgIT09IGEgJiYgYiAhPT0gYik7XG59XG5mdW5jdGlvbiAkJGVxdWFsczIoY2FjaGUsIGluZGV4LCBiKSB7XG4gIHJldHVybiBpbmRleCBpbiBjYWNoZSAmJiBpc0VxdWFsKGNhY2hlW2luZGV4XSwgYik7XG59XG5cbmNvbnN0IGNhY2hlID0gWydhJ10iLCJ0ZXN0cyI6W3sibmFtZSI6IlRlc3QgQ2FzZSIsImNvZGUiOiIkJGVxdWFsczIoY2FjaGUsIDEsICdhJyk7XG4kJGVxdWFsczIoY2FjaGUsIDEsICdiJyk7XG4kJGVxdWFsczIoY2FjaGUsIDIsICdhJyk7IiwicnVucyI6WzIwNjMwMDAsMTE0MTAwMCw0NjEwMDAsMTQ0ODAwMCw5MTcwMDAsMzExNjAwMCwzMzcxMDAwLDE4NTcwMDAsMTEwMDAsODAwMCwxODM5MDAwLDIzMDMwMDAsMzA5MDAwLDI4MjUwMDAsMzY2MDAwLDIwNjMwMDAsMTAwMCwyODI1MDAwLDE4ODAwMCwyMDMzMDAwLDE3MTgwMDAsODkxMDAwLDI1MzYwMDAsNzE0MDAwLDI4MjUwMDAsMzU1MDAwLDE1NzgwMDAsMjgyNTAwMCwzMjAwMCwxODcwMDAwLDE5NTAwMCw0MTkwMDAsNDgxMDAwLDQxMDAwMCwyMTY3MDAwLDQwMjAwMCwxNDAwMCw0ODYwMDAsNDQwMDAsMjI5MDAwLDIwMzkwMDAsMjQ2OTAwMCwxNDUwMDAsMTEyMzAwMCwzMTEwMDAwLDE0MzkwMDAsMTM1MjAwMCwyMjExMDAwLDE2OTAwMCwxMTIwMDAwLDI5NDAwMDAsMTIwMDAwMCwxMjA0MDAwLDQzMjAwMCwxMTUwMDAwLDcxMDAwMCw2NTcwMDAsNjk5MDAwLDE5NDQwMDAsMTQ1MTAwMCwyNjM5MDAwLDIzMzQwMDAsMTU2NjAwMCwxOTQxMDAwLDUxNjAwMCwxODEzMDAwLDgwMDAsMTM1NDAwMCwxNjAwMDAsMjg5MzAwMCw5MDAwLDQ5NzAwMCw0NTAwMDAsMTYzNjAwMCwyMjQwMDAsMjM0NDAwMCwzMTY3MDAwLDM0MzEwMDAsMTMzMDAwMCwxMjQ3MDAwLDEzNTIwMDAsMjM1MDAwLDQxMTMwMDAsMjY1NDAwMCwyNTk4MDAwLDI1MDAwMCwyMTUwMDAsMTMzNzAwMCwxNjA3MDAwLDM2NjcwMDAsMTcwMTAwMCwzNzg5MDAwLDEyNjgwMDAsNTEwMDAwLDk5OTAwMCwyMzM4MDAwLDc1MDAwMCwzMjQxMDAwLDM0ODcwMDAsMjE0MTAwMF0sIm9wcyI6MTQ0NzExMH0seyJuYW1lIjoiVGVzdCBDYXNlIiwiY29kZSI6IiQkZXF1YWxzMShjYWNoZSwgMSwgJ2EnKTtcbiQkZXF1YWxzMShjYWNoZSwgMSwgJ2InKTtcbiQkZXF1YWxzMShjYWNoZSwgMiwgJ2EnKTsiLCJydW5zIjpbMTkyMTAwMCwzMDU5MDAwLDQ2MTAwMCw4NjEwMDAsMjUxNTAwMCw4MzUwMDAsNDc5MDAwLDU5NTAwMCwxNjk4MDAwLDM2NjcwMDAsOTg1MDAwLDQyMTgwMDAsMzQ4NjAwMCwyNzg4MDAwLDc3MzAwMCw4NzAwMDAsMTE1ODAwMCwxMDAwLDEyMjEwMDAsMjMyNDAwMCwxOTk3MDAwLDIwNTUwMDAsMTYxNjAwMCw2ODkwMDAsNjkxMDAwLDIzNTkwMDAsNjU3MDAwLDIwNzgwMDAsMTM1MDAwMCw1OTUwMDAsMjEyMTAwMCwyNTE1MDAwLDMyOTgwMDAsMzk3MDAwLDIxMDUwMDAsMTk0MTAwMCwxNDAwMCwyNzQ3MDAwLDM2NDcwMDAsMjA0MzAwMCwxMDAwLDEzNjYwMDAsMjk5MDAwMCwxMTIxMDAwLDI1MTUwMDAsMTQyMDAwMCw0NTYwMDAsNDkwMDAwLDE1MjkwMDAsMTA5NzAwMCwxOTc2MDAwLDY5MDAwLDYzODAwMCwyNTQzMDAwLDc4NDAwMCwyMTcxMDAwLDE3NDEwMDAsMjUxNTAwMCwxNzY5MDAwLDE5MDYwMDAsMjUxNTAwMCwxMjMwMDAwLDE1MjAwMDAsNjMxMDAwLDE2NDQwMDAsNzYzMDAwLDEwOTQwMDAsMTIwMDAsMTY4MDAwMCwyMDM5MDAwLDE3NzYwMDAsNDk0MDAwLDIyNDQwMDAsNDkwMDAsMzYzMDAwLDYwNTAwMCw2OTAwMDAsMTgzMTAwMCwxMDAwLDU5OTAwMCw4NDUwMDAsMzAwOTAwMCw0MDY1MDAwLDE5MTQwMDAsMTM4MjAwMCw3NTEwMDAsMzM3NzAwMCw0MDQwMDAsNjkwMDAsMjM5MTAwMCw0OTMwMDAsMjUxNTAwMCwxMTUyMDAwLDE1NTAwMDAsMTAwMCwxMjIyMDAwLDM0MzEwMDAsMjY0MzAwMCwzNTI2MDAwLDU0ODAwMF0sIm9wcyI6MTU0OTk1MH1dLCJ1cGRhdGVkIjoiMjAyMy0wNS0xN1QwNDoxNTozMC44MjRaIn0%3D).

IMHO we should keep this.

Reply:
I'm using `isEqual` in another form and possibly more in the future. `isEqual` is going to be another hot path so it's fine.